test/structuring_element.jl

```diff
@@ -0,0 +1,260 @@
+@testset "strel" begin
+    @testset "offset to mask" begin
+        se = [CartesianIndex(-1), CartesianIndex(1)]
+        se_mask = @inferred strel(Bool, se)
+        @test se_mask == Bool[1, 1, 1]
+
+        se = [CartesianIndex(-1, -1), CartesianIndex(1, 1)]
+        @test strel(se) === se
+        se_mask = @inferred strel(Bool, se)
+        @test se_mask == Bool[1 0 0; 0 1 0; 0 0 1]
+        # internally, Bool is converted to SEMask
+        se_mask = @inferred strel(ImageMorphology.SEMask{2}(), se)
+        @test se_mask == Bool[1 0 0; 0 1 0; 0 0 1]
+
+        se = CartesianIndices((-1:1, -1:1))
+        se_mask = @inferred strel(Bool, se)
+        @test se_mask == trues((3, 3))
+
+        se = CartesianIndices((1:2, -1:1))
+        se_mask = @inferred strel(Bool, se)
+        @test se_mask == Bool[0 0 0; 0 0 0; 0 1 0; 1 1 1; 1 1 1]
+
+        se = @inferred strel(Bool, CartesianIndex{2}[])
+        @test se == reshape(Bool[1], (1, 1)) # v1.7: Bool[1;;]
+
+        # invalid cases
+        se = [CartesianIndex(1), CartesianIndex(1, 1)]
+        @test_throws ErrorException strel(se)
+    end
+
+    @testset "mask to offset" begin
+        se = Bool[1, 0, 1]
+        se_offsets = if VERSION >= v"1.6"
+            @inferred strel(CartesianIndex, se)
+        else
+            strel(CartesianIndex, se)
+        end
+        @test se_offsets == [CartesianIndex(-1,), CartesianIndex(1,)]
+        @test se_offsets == strel(CartesianIndex, Bool[1, 1, 1])
+
+        se = Bool[1 0 0; 0 1 0; 0 0 1]
+        se_offsets = if VERSION >= v"1.6"
+            @inferred strel(CartesianIndex, se)
+        else
+            strel(CartesianIndex, se)
+        end
+        @test se_offsets == [CartesianIndex(-1, -1), CartesianIndex(1, 1)]
+        # internally, CartesianIndex is converted to SEOffsets
+        se_offsets = if VERSION >= v"1.6"
+            @inferred strel(ImageMorphology.SEOffsets{2}(), se)
+        else
+            strel(ImageMorphology.SEOffsets{2}(), se)
+        end
+        @test se_offsets == [CartesianIndex(-1, -1), CartesianIndex(1, 1)]
+
+        se = trues((3, 3))
+        se_offsets = if VERSION >= v"1.6"
+            @inferred strel(CartesianIndex, se)
+        else
+            strel(CartesianIndex, se)
+        end
+        @test se_offsets == filter(x -> !iszero(x), vec(CartesianIndices((-1:1, -1:1))))
+
+        se = if VERSION >= v"1.6"
+            @inferred strel(CartesianIndex, falses(3, 3))
+        else
+            strel(CartesianIndex, falses(3, 3))
+        end
+        @test isempty(se)
+    end
+
+    @testset "center point" begin
+        for se in [Bool[1 0 0; 0 1 0; 0 0 1], Bool[1 0 0; 0 0 0; 0 0 1]]
+            # center point is always excluded in offsets
```

Comment:
Why? the list of offset could contains the center, sometimes we use it sometimes not